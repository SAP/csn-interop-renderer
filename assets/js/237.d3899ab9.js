"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[237],{2237:(__unused_webpack_module,__webpack_exports__,__webpack_require__)=>{eval('__webpack_require__.r(__webpack_exports__);\n/* harmony export */ __webpack_require__.d(__webpack_exports__, {\n/* harmony export */   "default": () => (/* binding */ Index)\n/* harmony export */ });\n/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(6540);\n/* harmony import */ var _docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(1312);\n/* harmony import */ var _docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_5__ = __webpack_require__(5500);\n/* harmony import */ var _theme_Layout__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(1656);\n/* harmony import */ var _theme_NotFound_Content__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(3363);\n/* harmony import */ var react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(4848);\n/**\n * Copyright (c) Facebook, Inc. and its affiliates.\n *\n * This source code is licensed under the MIT license found in the\n * LICENSE file in the root directory of this source tree.\n */function Index(){const title=(0,_docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__/* .translate */ .T)({id:\'theme.NotFound.title\',message:\'Page Not Found\'});return/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__.jsxs)(react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__.Fragment,{children:[/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__.jsx)(_docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_5__/* .PageMetadata */ .be,{title:title}),/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__.jsx)(_theme_Layout__WEBPACK_IMPORTED_MODULE_2__/* ["default"] */ .A,{children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_4__.jsx)(_theme_NotFound_Content__WEBPACK_IMPORTED_MODULE_3__/* ["default"] */ .A,{})})]});}//# sourceURL=[module]\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiMjIzNy5qcyIsIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7O0FBQUE7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBLEdBTWUsUUFBUyxDQUFBVyxLQUFLQSxDQUFBLENBQUcsQ0FDOUIsS0FBTSxDQUFBQyxLQUFLLENBQUdYLHlFQUFTLENBQUMsQ0FDdEJZLEVBQUUsQ0FBRSxzQkFBc0IsQ0FDMUJDLE9BQU8sQ0FBRSxnQkFDWCxDQUFDLENBQUMsQ0FDRixtQkFDRUosdURBQUEsQ0FBQUYsdURBQUEsRUFBQU8sUUFBQSxlQUNFVCxzREFBQSxDQUFDSiw0RUFBWSxFQUFDVSxLQUFLLENBQUVBLEtBQU0sQ0FBRSxDQUFDLGNBQzlCTixzREFBQSxDQUFDSCw4REFBTSxFQUFBWSxRQUFBLGNBQ0xULHNEQUFBLENBQUNGLHdFQUFlLEdBQUUsQ0FBQyxDQUNiLENBQUMsRUFDVCxDQUFDLENBRVAiLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly9Ac2FwL2Nzbi1pbnRlcm9wLXJlbmRlcmVyLXdlYnNpdGUvLi9ub2RlX21vZHVsZXMvQGRvY3VzYXVydXMvdGhlbWUtY2xhc3NpYy9saWIvdGhlbWUvTm90Rm91bmQvaW5kZXguanM/MTkwNiJdLCJzb3VyY2VzQ29udGVudCI6WyIvKipcbiAqIENvcHlyaWdodCAoYykgRmFjZWJvb2ssIEluYy4gYW5kIGl0cyBhZmZpbGlhdGVzLlxuICpcbiAqIFRoaXMgc291cmNlIGNvZGUgaXMgbGljZW5zZWQgdW5kZXIgdGhlIE1JVCBsaWNlbnNlIGZvdW5kIGluIHRoZVxuICogTElDRU5TRSBmaWxlIGluIHRoZSByb290IGRpcmVjdG9yeSBvZiB0aGlzIHNvdXJjZSB0cmVlLlxuICovXG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHt0cmFuc2xhdGV9IGZyb20gJ0Bkb2N1c2F1cnVzL1RyYW5zbGF0ZSc7XG5pbXBvcnQge1BhZ2VNZXRhZGF0YX0gZnJvbSAnQGRvY3VzYXVydXMvdGhlbWUtY29tbW9uJztcbmltcG9ydCBMYXlvdXQgZnJvbSAnQHRoZW1lL0xheW91dCc7XG5pbXBvcnQgTm90Rm91bmRDb250ZW50IGZyb20gJ0B0aGVtZS9Ob3RGb3VuZC9Db250ZW50JztcbmV4cG9ydCBkZWZhdWx0IGZ1bmN0aW9uIEluZGV4KCkge1xuICBjb25zdCB0aXRsZSA9IHRyYW5zbGF0ZSh7XG4gICAgaWQ6ICd0aGVtZS5Ob3RGb3VuZC50aXRsZScsXG4gICAgbWVzc2FnZTogJ1BhZ2UgTm90IEZvdW5kJyxcbiAgfSk7XG4gIHJldHVybiAoXG4gICAgPD5cbiAgICAgIDxQYWdlTWV0YWRhdGEgdGl0bGU9e3RpdGxlfSAvPlxuICAgICAgPExheW91dD5cbiAgICAgICAgPE5vdEZvdW5kQ29udGVudCAvPlxuICAgICAgPC9MYXlvdXQ+XG4gICAgPC8+XG4gICk7XG59XG4iXSwibmFtZXMiOlsiUmVhY3QiLCJ0cmFuc2xhdGUiLCJQYWdlTWV0YWRhdGEiLCJMYXlvdXQiLCJOb3RGb3VuZENvbnRlbnQiLCJqc3giLCJfanN4IiwiRnJhZ21lbnQiLCJfRnJhZ21lbnQiLCJqc3hzIiwiX2pzeHMiLCJJbmRleCIsInRpdGxlIiwiaWQiLCJtZXNzYWdlIiwiY2hpbGRyZW4iXSwic291cmNlUm9vdCI6IiJ9\n//# sourceURL=webpack-internal:///2237\n')},3363:(__unused_webpack_module,__webpack_exports__,__webpack_require__)=>{eval('/* harmony export */ __webpack_require__.d(__webpack_exports__, {\n/* harmony export */   A: () => (/* binding */ NotFoundContent)\n/* harmony export */ });\n/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(6540);\n/* harmony import */ var clsx__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(4164);\n/* harmony import */ var _docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(1312);\n/* harmony import */ var _theme_Heading__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(1107);\n/* harmony import */ var react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(4848);\n/**\n * Copyright (c) Facebook, Inc. and its affiliates.\n *\n * This source code is licensed under the MIT license found in the\n * LICENSE file in the root directory of this source tree.\n */function NotFoundContent({className}){return/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)("main",{className:(0,clsx__WEBPACK_IMPORTED_MODULE_4__/* ["default"] */ .A)(\'container margin-vert--xl\',className),children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)("div",{className:"row",children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsxs)("div",{className:"col col--6 col--offset-3",children:[/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)(_theme_Heading__WEBPACK_IMPORTED_MODULE_2__/* ["default"] */ .A,{as:"h1",className:"hero__title",children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)(_docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__/* ["default"] */ .A,{id:"theme.NotFound.title",description:"The title of the 404 page",children:"Page Not Found"})}),/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)("p",{children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)(_docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__/* ["default"] */ .A,{id:"theme.NotFound.p1",description:"The first paragraph of the 404 page",children:"We could not find what you were looking for."})}),/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)("p",{children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_3__.jsx)(_docusaurus_Translate__WEBPACK_IMPORTED_MODULE_1__/* ["default"] */ .A,{id:"theme.NotFound.p2",description:"The 2nd paragraph of the 404 page",children:"Please contact the owner of the site that linked you to the original URL and let them know their link is broken."})})]})})});}//# sourceURL=[module]\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiMzM2My5qcyIsIm1hcHBpbmdzIjoiOzs7Ozs7OztBQUFBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQSxHQUtlLFFBQVMsQ0FBQVEsZUFBZUEsQ0FBQyxDQUFDQyxTQUFTLENBQUMsQ0FBRSxDQUNuRCxtQkFDRUosc0RBQUEsU0FBTUksU0FBUyxDQUFFUix5REFBSSxDQUFDLDJCQUEyQixDQUFFUSxTQUFTLENBQUUsQ0FBQUMsUUFBQSxjQUM1REwsc0RBQUEsUUFBS0ksU0FBUyxDQUFDLEtBQUssQ0FBQUMsUUFBQSxjQUNsQkgsdURBQUEsUUFBS0UsU0FBUyxDQUFDLDBCQUEwQixDQUFBQyxRQUFBLGVBQ3ZDTCxzREFBQSxDQUFDRiwrREFBTyxFQUFDUSxFQUFFLENBQUMsSUFBSSxDQUFDRixTQUFTLENBQUMsYUFBYSxDQUFBQyxRQUFBLGNBQ3RDTCxzREFBQSxDQUFDSCxzRUFBUyxFQUNSVSxFQUFFLENBQUMsc0JBQXNCLENBQ3pCQyxXQUFXLENBQUMsMkJBQTJCLENBQUFILFFBQUEsQ0FBQyxnQkFFMUMsQ0FBVyxDQUFDLENBQ0wsQ0FBQyxjQUNWTCxzREFBQSxNQUFBSyxRQUFBLGNBQ0VMLHNEQUFBLENBQUNILHNFQUFTLEVBQ1JVLEVBQUUsQ0FBQyxtQkFBbUIsQ0FDdEJDLFdBQVcsQ0FBQyxxQ0FBcUMsQ0FBQUgsUUFBQSxDQUFDLDhDQUVwRCxDQUFXLENBQUMsQ0FDWCxDQUFDLGNBQ0pMLHNEQUFBLE1BQUFLLFFBQUEsY0FDRUwsc0RBQUEsQ0FBQ0gsc0VBQVMsRUFDUlUsRUFBRSxDQUFDLG1CQUFtQixDQUN0QkMsV0FBVyxDQUFDLG1DQUFtQyxDQUFBSCxRQUFBLENBQUMsa0hBR2xELENBQVcsQ0FBQyxDQUNYLENBQUMsRUFDRCxDQUFDLENBQ0gsQ0FBQyxDQUNGLENBQUMsQ0FFWCIsInNvdXJjZXMiOlsid2VicGFjazovL0BzYXAvY3NuLWludGVyb3AtcmVuZGVyZXItd2Vic2l0ZS8uL25vZGVfbW9kdWxlcy9AZG9jdXNhdXJ1cy90aGVtZS1jbGFzc2ljL2xpYi90aGVtZS9Ob3RGb3VuZC9Db250ZW50L2luZGV4LmpzP2I2NDkiXSwic291cmNlc0NvbnRlbnQiOlsiLyoqXG4gKiBDb3B5cmlnaHQgKGMpIEZhY2Vib29rLCBJbmMuIGFuZCBpdHMgYWZmaWxpYXRlcy5cbiAqXG4gKiBUaGlzIHNvdXJjZSBjb2RlIGlzIGxpY2Vuc2VkIHVuZGVyIHRoZSBNSVQgbGljZW5zZSBmb3VuZCBpbiB0aGVcbiAqIExJQ0VOU0UgZmlsZSBpbiB0aGUgcm9vdCBkaXJlY3Rvcnkgb2YgdGhpcyBzb3VyY2UgdHJlZS5cbiAqL1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBjbHN4IGZyb20gJ2Nsc3gnO1xuaW1wb3J0IFRyYW5zbGF0ZSBmcm9tICdAZG9jdXNhdXJ1cy9UcmFuc2xhdGUnO1xuaW1wb3J0IEhlYWRpbmcgZnJvbSAnQHRoZW1lL0hlYWRpbmcnO1xuZXhwb3J0IGRlZmF1bHQgZnVuY3Rpb24gTm90Rm91bmRDb250ZW50KHtjbGFzc05hbWV9KSB7XG4gIHJldHVybiAoXG4gICAgPG1haW4gY2xhc3NOYW1lPXtjbHN4KCdjb250YWluZXIgbWFyZ2luLXZlcnQtLXhsJywgY2xhc3NOYW1lKX0+XG4gICAgICA8ZGl2IGNsYXNzTmFtZT1cInJvd1wiPlxuICAgICAgICA8ZGl2IGNsYXNzTmFtZT1cImNvbCBjb2wtLTYgY29sLS1vZmZzZXQtM1wiPlxuICAgICAgICAgIDxIZWFkaW5nIGFzPVwiaDFcIiBjbGFzc05hbWU9XCJoZXJvX190aXRsZVwiPlxuICAgICAgICAgICAgPFRyYW5zbGF0ZVxuICAgICAgICAgICAgICBpZD1cInRoZW1lLk5vdEZvdW5kLnRpdGxlXCJcbiAgICAgICAgICAgICAgZGVzY3JpcHRpb249XCJUaGUgdGl0bGUgb2YgdGhlIDQwNCBwYWdlXCI+XG4gICAgICAgICAgICAgIFBhZ2UgTm90IEZvdW5kXG4gICAgICAgICAgICA8L1RyYW5zbGF0ZT5cbiAgICAgICAgICA8L0hlYWRpbmc+XG4gICAgICAgICAgPHA+XG4gICAgICAgICAgICA8VHJhbnNsYXRlXG4gICAgICAgICAgICAgIGlkPVwidGhlbWUuTm90Rm91bmQucDFcIlxuICAgICAgICAgICAgICBkZXNjcmlwdGlvbj1cIlRoZSBmaXJzdCBwYXJhZ3JhcGggb2YgdGhlIDQwNCBwYWdlXCI+XG4gICAgICAgICAgICAgIFdlIGNvdWxkIG5vdCBmaW5kIHdoYXQgeW91IHdlcmUgbG9va2luZyBmb3IuXG4gICAgICAgICAgICA8L1RyYW5zbGF0ZT5cbiAgICAgICAgICA8L3A+XG4gICAgICAgICAgPHA+XG4gICAgICAgICAgICA8VHJhbnNsYXRlXG4gICAgICAgICAgICAgIGlkPVwidGhlbWUuTm90Rm91bmQucDJcIlxuICAgICAgICAgICAgICBkZXNjcmlwdGlvbj1cIlRoZSAybmQgcGFyYWdyYXBoIG9mIHRoZSA0MDQgcGFnZVwiPlxuICAgICAgICAgICAgICBQbGVhc2UgY29udGFjdCB0aGUgb3duZXIgb2YgdGhlIHNpdGUgdGhhdCBsaW5rZWQgeW91IHRvIHRoZVxuICAgICAgICAgICAgICBvcmlnaW5hbCBVUkwgYW5kIGxldCB0aGVtIGtub3cgdGhlaXIgbGluayBpcyBicm9rZW4uXG4gICAgICAgICAgICA8L1RyYW5zbGF0ZT5cbiAgICAgICAgICA8L3A+XG4gICAgICAgIDwvZGl2PlxuICAgICAgPC9kaXY+XG4gICAgPC9tYWluPlxuICApO1xufVxuIl0sIm5hbWVzIjpbIlJlYWN0IiwiY2xzeCIsIlRyYW5zbGF0ZSIsIkhlYWRpbmciLCJqc3giLCJfanN4IiwianN4cyIsIl9qc3hzIiwiTm90Rm91bmRDb250ZW50IiwiY2xhc3NOYW1lIiwiY2hpbGRyZW4iLCJhcyIsImlkIiwiZGVzY3JpcHRpb24iXSwic291cmNlUm9vdCI6IiJ9\n//# sourceURL=webpack-internal:///3363\n')}}]);