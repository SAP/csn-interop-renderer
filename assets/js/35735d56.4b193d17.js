"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[775],{2082:e=>{e.exports=JSON.parse('{"version":{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"defaultSidebar":[{"type":"link","label":"Overview","href":"/csn-interop-renderer/","docId":"index","unlisted":false},{"type":"category","label":"Integrations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"NPM package","href":"/csn-interop-renderer/Integrations/npm","docId":"Integrations/npm","unlisted":false},{"type":"link","label":"Web Component","href":"/csn-interop-renderer/Integrations/web-component","docId":"Integrations/web-component","unlisted":false},{"type":"link","label":"Using CLI","href":"/csn-interop-renderer/Integrations/cli","docId":"Integrations/cli","unlisted":false}]},{"type":"category","label":"examples","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Airline (Example)","href":"/csn-interop-renderer/examples/Airline","docId":"examples/Airline","unlisted":false},{"type":"link","label":"BillingDocument","href":"/csn-interop-renderer/examples/BillingDocument","docId":"examples/BillingDocument","unlisted":false},{"type":"link","label":"CostCenter","href":"/csn-interop-renderer/examples/CostCenter","docId":"examples/CostCenter","unlisted":false},{"type":"link","label":"CostCenterCombined","href":"/csn-interop-renderer/examples/CostCenterCombined","docId":"examples/CostCenterCombined","unlisted":false},{"type":"link","label":"GeneralLedgerAccountLineItem","href":"/csn-interop-renderer/examples/GeneralLedgerAccountLineItem","docId":"examples/GeneralLedgerAccountLineItem","unlisted":false}]}]},"docs":{"examples/Airline":{"id":"examples/Airline","title":"Airline (Example)","description":"This CSN example document shows how the airline example is expressed with a CDS Service exposing the entities through an API.","sidebar":"defaultSidebar"},"examples/BillingDocument":{"id":"examples/BillingDocument","title":"BillingDocument","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/CostCenter":{"id":"examples/CostCenter","title":"CostCenter","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/CostCenterCombined":{"id":"examples/CostCenterCombined","title":"CostCenterCombined","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/GeneralLedgerAccountLineItem":{"id":"examples/GeneralLedgerAccountLineItem","title":"GeneralLedgerAccountLineItem","description":"Entity Definitions","sidebar":"defaultSidebar"},"index":{"id":"index","title":"Overview","description":"Motivation","sidebar":"defaultSidebar"},"Integrations/cli":{"id":"Integrations/cli","title":"Using CLI","description":"Suitable for local testing on your machine.","sidebar":"defaultSidebar"},"Integrations/npm":{"id":"Integrations/npm","title":"NPM package","description":"Suitable for JavaScript/TypeScript based projects.","sidebar":"defaultSidebar"},"Integrations/web-component":{"id":"Integrations/web-component","title":"Web Component","description":"Suitable for Frontend e.g. React/Angular based projects.","sidebar":"defaultSidebar"}}}}')}}]);