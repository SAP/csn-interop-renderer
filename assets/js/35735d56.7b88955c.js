"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[775],{2082:e=>{e.exports=JSON.parse('{"version":{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"defaultSidebar":[{"type":"link","label":"Overview","href":"/csn-interop-renderer/","docId":"index","unlisted":false},{"type":"category","label":"Integrations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"NPM package","href":"/csn-interop-renderer/Integrations/npm","docId":"Integrations/npm","unlisted":false},{"type":"link","label":"Web Component","href":"/csn-interop-renderer/Integrations/web-component","docId":"Integrations/web-component","unlisted":false},{"type":"link","label":"Using CLI","href":"/csn-interop-renderer/Integrations/cli","docId":"Integrations/cli","unlisted":false}]},{"type":"category","label":"Renderer examples","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Airline","href":"/csn-interop-renderer/Renderer examples/Airline","docId":"Renderer examples/Airline","unlisted":false},{"type":"link","label":"BillingDocument","href":"/csn-interop-renderer/Renderer examples/BillingDocument","docId":"Renderer examples/BillingDocument","unlisted":false},{"type":"link","label":"CostCenter","href":"/csn-interop-renderer/Renderer examples/CostCenter","docId":"Renderer examples/CostCenter","unlisted":false},{"type":"link","label":"CostCenterCombined","href":"/csn-interop-renderer/Renderer examples/CostCenterCombined","docId":"Renderer examples/CostCenterCombined","unlisted":false},{"type":"link","label":"GeneralLedgerAccountLineItem","href":"/csn-interop-renderer/Renderer examples/GeneralLedgerAccountLineItem","docId":"Renderer examples/GeneralLedgerAccountLineItem","unlisted":false}]},{"type":"link","label":"Support","href":"/csn-interop-renderer/support","docId":"support","unlisted":false}]},"docs":{"index":{"id":"index","title":"Overview","description":"Motivation","sidebar":"defaultSidebar"},"Integrations/cli":{"id":"Integrations/cli","title":"Using CLI","description":"Suitable for local testing on your machine.","sidebar":"defaultSidebar"},"Integrations/npm":{"id":"Integrations/npm","title":"NPM package","description":"Suitable for JavaScript/TypeScript based projects.","sidebar":"defaultSidebar"},"Integrations/web-component":{"id":"Integrations/web-component","title":"Web Component","description":"Suitable for Frontend e.g. React/Angular based projects.","sidebar":"defaultSidebar"},"Renderer examples/Airline":{"id":"Renderer examples/Airline","title":"Airline","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/BillingDocument":{"id":"Renderer examples/BillingDocument","title":"BillingDocument","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/CostCenter":{"id":"Renderer examples/CostCenter","title":"CostCenter","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/CostCenterCombined":{"id":"Renderer examples/CostCenterCombined","title":"CostCenterCombined","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/GeneralLedgerAccountLineItem":{"id":"Renderer examples/GeneralLedgerAccountLineItem","title":"GeneralLedgerAccountLineItem","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"support":{"id":"support","title":"Support","description":"Reporting Bugs","sidebar":"defaultSidebar"}}}}')}}]);