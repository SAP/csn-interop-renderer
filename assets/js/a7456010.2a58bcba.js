"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[235],{8552:e=>{e.exports=JSON.parse('{"name":"docusaurus-plugin-content-pages","id":"default"}')}}]);