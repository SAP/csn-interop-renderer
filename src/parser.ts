import { CSNInteropRoot, schemas } from "@sap/csn-interop-specification";
import { renderer } from "./renderer.js";
import { CsnInteropRendererConfig } from "./types/index.js";

export const parser = async (
  text: CSNInteropRoot,
  config?: CsnInteropRendererConfig,
  generateAsHTml: boolean = false,
): Promise<string> => {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  let Draft07;
  const jsonSchemaLib = await import("json-schema-library");

  // Check if we are in Node environment
  if (typeof process !== "undefined" && process.versions && process.versions.node) {
    // Node environment
    Draft07 = jsonSchemaLib.default.Draft07;
  } else {
    // Browser environment
    Draft07 = jsonSchemaLib.Draft07;
  }

  const draft07 = new Draft07(schemas.csnInteropEffectiveSchema);
  const errors = draft07.validate(text);

  if (errors.length) {
    const preparedErrors = [];

    for (const error of errors) {
      preparedErrors.push({ message: error.message, type: error.type, code: error.code });
    }

    throw new Error(
      preparedErrors.reduce((combinedMessage, error) => {
        combinedMessage += `\n[${error.type}]: ${error.message} (${error.code})`;
        return combinedMessage;
      }, ""),
    );
  }
  return renderer(text, config, generateAsHTml);
};
