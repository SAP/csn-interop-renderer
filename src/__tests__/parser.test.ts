import { parser } from "../parser.js";
import example from "../../examples/CSNInterop.js";
import { CSNInteropRoot } from "@sap/csn-interop-specification";

describe("Parser Tests", () => {
  test("should throw some errors if corrupt file is provided", async () => {
    const exampleString = "asdasjklajs;;";
    expect.assertions(2);

    try {
      await parser(exampleString as unknown as CSNInteropRoot);
    } catch (error: unknown) {
      expect(error).toBeDefined();
      expect(error).toMatchSnapshot();
    }
  });

  test("should throw some errors JSON file is invalid", async () => {
    // @ts-expect-error We intentionally break something here
    delete example.definitions;

    expect.assertions(2);

    try {
      await parser(example);
    } catch (error: unknown) {
      expect(error).toBeDefined();
      expect(error).toMatchSnapshot();
    }
  });
});
