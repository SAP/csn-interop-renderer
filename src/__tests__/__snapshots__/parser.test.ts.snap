// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Tests should throw some errors JSON file is invalid 1`] = `
[Error: 
[error]: The required property \`definitions\` is missing at \`#\` (required-property-error)]
`;

exports[`Parser Tests should throw some errors if corrupt file is provided 1`] = `
[Error: 
[error]: Expected \`asdasjklajs;;\` (string) in \`#\` to be of type \`object\` (type-error)]
`;
